Bottom: 163bbe971bfc8d32e2434473189ccf9646b7bc81
Top:    a00c6d2645cef8185d0adc91c2b142e34602388e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2015-03-23 00:13:02 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 5cd245f..c5cab7d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
